### PR TITLE
Infra fixes for bugs uncovered by testing of the JIT/Methodical merged tests

### PR DESCRIPTION
**Pull request by trylek**
This is a series of changes made by Jeremy and myself based on our observations from lab testing
of the JIT/Methodical switchover to use merged test wrappers. They deal with various previously
unseen special cases like passing payloads for out-of-process tests to Helix, proper propagation
of test exclusion list (issues.targets) to the generated merged wrapper and multiple fixes for Mono.

Thanks

Tomas

### CODE REVIEW

File: src/tests/Common/testgrouping.proj [hunk 1]
```diff
@@ -180,42 +180,6 @@
       <XUnitWrapperDll>$(TestBinDir)JIT\HardwareIntrinsics\JIT.HardwareIntrinsics.XUnitWrapper.dll</XUnitWrapperDll>
     </TestGrouping>
 
-    <TestGrouping Include="$(TestBinDir)JIT\Methodical\a*\**;
```
Comment (jkoritzinsky) [inline src/tests/Common/testgrouping.proj:183]: We should undo the changes in this file. This change is only for when we actually merge the JIT/Methodical work.

**Review by jkoritzinsky — CHANGES_REQUESTED** [review]

File: src/tests/Common/testgrouping.proj (hunk 1 above)
Comment (trylek) [reply src/tests/Common/testgrouping.proj]: Thanks for pointing that out, reverted in the latest commit.

**Review by jkoritzinsky — APPROVED** [review]

File: src/tests/Loader/classloader/MethodImpl/CovariantReturns/UnitTest/UnitTestMultiModule.ilproj [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk.IL">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
+    <MonoAotIncompatible>true</MonoAotIncompatible>
```
Comment (jkoritzinsky) [inline src/tests/Loader/classloader/MethodImpl/CovariantReturns/UnitTest/UnitTestMultiModule.ilproj:4]:
```suggestion
```

Marking this as incompatible is what is causing the test failures. We'll revisit how to handle this better later.

File: src/tests/Loader/classloader/generics/Variance/IL/Unbox002.csproj [hunk 3]
```diff
@@ -2,6 +2,7 @@
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <OutputType>Exe</OutputType>
+    <MonoAotIncompatible>true</MonoAotIncompatible>
```
Comment (jkoritzinsky) [inline src/tests/Loader/classloader/generics/Variance/IL/Unbox002.csproj:5]:
```suggestion
```

File: src/tests/Loader/classloader/generics/Variance/IL/Unbox003.csproj [hunk 4]
```diff
@@ -2,6 +2,7 @@
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <OutputType>Exe</OutputType>
+    <MonoAotIncompatible>true</MonoAotIncompatible>
```
Comment (jkoritzinsky) [inline src/tests/Loader/classloader/generics/Variance/IL/Unbox003.csproj:5]:
```suggestion
```

File: src/tests/Loader/classloader/generics/Variance/IL/vsw543506.csproj [hunk 5]
```diff
@@ -2,6 +2,7 @@
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <OutputType>Exe</OutputType>
+    <MonoAotIncompatible>true</MonoAotIncompatible>
```
Comment (jkoritzinsky) [inline src/tests/Loader/classloader/generics/Variance/IL/vsw543506.csproj:5]:
```suggestion
```
